src/test/java/org/sasanlabs/service/vulnerability/commandInjection/CommandInjectionTest.java

```diff
@@ -0,0 +1,76 @@
+package org.sasanlabs.service.vulnerability.commandInjection;
+
+import java.io.IOException;
+import java.net.URI;
+import org.junit.jupiter.api.Test;
+import org.sasanlabs.service.exception.ServiceApplicationException;
+import org.sasanlabs.service.vulnerability.bean.GenericVulnerabilityResponseBean;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.Assert;
+
+public class CommandInjectionTest {
+
+    CommandInjection commandInjection = new CommandInjection();
+    private String wrongUrl = "localhsost";
+
+    @Test
+    public void getVulnerablePayloadLevel1_Localhost_ExpectPingSuccess() throws IOException {
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel1("127.0.0.1");
+        Assert.isTrue(
+                responseEntity
+                        .getBody()
+                        .getContent()
+                        .startsWith("PING 127.0.0.1 (127.0.0.1): 56 data bytes"),
+                "must get response");
+        Assert.isTrue(HttpStatus.OK == responseEntity.getStatusCode(), "must equal");
+        Assert.isTrue(responseEntity.getBody().getIsValid(), "must equal");
+    }
+
```

Comment:
@fengyuanyang I doubt this test would run ,
So the `getVulnerablePayloadLevel1()` method is actually a `RequestMapping` and the `CommandInjection`  class is a `RestController` basically what that means is the class and the method together expose a REST endpoint , 
So the test should focus on checking whether this endpoint can be reached when it runs in a test server. So that would require using a `TestRunner` 
I would recommend giving https://spring.io/guides/gs/testing-web/ a read .

Reply:
I think we need not to spawn a server as per my understanding lets only stick to business logic testing and mock all these OS calls.

Reply:
IMHO RestController tests should be SprintBootTests only , their sole purpose would be to check if the method as an endpoint is reachable , 
other method calls like `getResponseFromPingCommand()` can be mocked for sanity.

Reply:
I would like to mock all these OS calls